### DISCUSSION TITLE
Creating overlay: How?

### DISCUSSION THREAD

**Discussion by joergRossdeutscher** [Helium314/SCEE#498] [Q&A] [answered]
Hello,
I tried to build my own overlay: 
buildings that don't have set roof:type yet.

1. I can enter three values, but I cannot find out which syntax / query language has to be used here. At least "building and !roof:type" doesn't throw an error, but also no results.
2. I cannot find out what is the "meaning" of the three input fields. Is there documentation?
3. When I (for learning purposes) enter just "building" in the first two fields I can identify multiple building types. But if I just enter "roof:type" in both fields nothing is happening. Is the double colon syntax wrong?
4. It very difficult to play around with the first input line. It seems to be syntax checked immediately, so a staccato of error message toasts is overlaying my keyboard after every single typed character, so I can't continue typing Is this a bug or another OnePlus "optimized user experience"?

Thanks for helping!

**Comment by Helium314** [comment, marked as answer]
> At least "building and !roof:type" doesn't throw an error, but also no results.

For me it works (highlights _all_ buildings, as `roof:type` has 0 uses)

> I cannot find out what is the "meaning" of the three input fields. Is there documentation?

You can click the large orange text for details. I added some info icon recently, but that will only show up in the next version.

Filtering details and dashed element filter use the same [syntax](https://github.com/Helium314/SCEE/blob/modified/CONTRIBUTING_A_NEW_QUEST.md#element-selection), the color filter uses regex.

> But if I just enter "roof:type" in both fields nothing is happening. Is the double colon syntax wrong?

`roof:type` has 0 uses, so you only will see anything if you enable _highlight missing data_ (will highlight everything that does _not_ match the color filter in red)

> It very difficult to play around with the first input line. It seems to be syntax checked immediately, so a staccato of error message toasts is overlaying my keyboard after every single typed character, so I can't continue typing Is this a bug or another OnePlus "optimized user experience"?

On my phone toasts are transparent enough to see the keyboard beneath, and do not block keyboard inputs. I guess this depends on the excat Android version.

**Comment by joergRossdeutscher** [comment]
Thanks for your feedback. I will fiddle around with that, but the main “issues“ obviously were… me.

1. I was so focussed on writing the query, that I didn't notice that “roof:type“ is wrong, and “roof:shape“ is what I want.
2. I didn't understand the red texts are links, so I never clicked them. They still looks like a headline to me, but maybe that's just me. However, obviously this is the documentation I was searching for.

Thanks a lot!
